packages/medusa/src/scripts/gift-card-tax-rate-migration.ts

```diff
@@ -0,0 +1,105 @@
+import dotenv from "dotenv"
+import { createConnection } from "typeorm"
+import Logger from "../loaders/logger"
+import { GiftCard } from "../models/gift-card"
+
+dotenv.config()
+
+const typeormConfig = {
+  type: process.env.TYPEORM_CONNECTION,
+  url: process.env.TYPEORM_URL,
+  username: process.env.TYPEORM_USERNAME,
+  password: process.env.TYPEORM_PASSWORD,
+  database: process.env.TYPEORM_DATABASE,
+  migrations: [process.env.TYPEORM_MIGRATIONS as string],
+  entities: [process.env.TYPEORM_ENTITIES],
+  logging: true,
+}
+
+const migrate = async function ({ typeormConfig }): Promise<void> {
+  const connection = await createConnection(typeormConfig)
+  const BATCH_SIZE = 1000
+  const migrationName = 'gift-card-tax-rate-migration'
+
+  await connection.transaction(async (manager) => {
+    let offset = 0
+
+    // Get all the GiftCards where the gift_card.tax_rate is null
+    const giftCardsCount = await manager
+      .createQueryBuilder()
+      .from(GiftCard, "gc")
+      .select("gc.id")
```

Comment:
**questions:**
- Should we also add a condition where region.gift_cards_are_taxable = true?
- Should we skip updating for those gift cards that have a gift card transaction already?